examples/callback/example_callback.cpp

```diff
@@ -0,0 +1,149 @@
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <ArborX.hpp>
+
+#include <Kokkos_Core.hpp>
+
+#include <iostream>
+#include <random>
+#include <vector>
+
+using ExecutionSpace = Kokkos::DefaultExecutionSpace;
+using MemorySpace = ExecutionSpace::memory_space;
+
+struct FirstOctant
+{
+};
+
+struct NearestToOrigin
+{
+  int k;
+};
+
+namespace ArborX
+{
+namespace Traits
+{
+template <>
+struct Access<FirstOctant, PredicatesTag>
+{
+  KOKKOS_FUNCTION static std::size_t size(FirstOctant) { return 1; }
+  KOKKOS_FUNCTION static auto get(FirstOctant, std::size_t)
+  {
+    return intersects(Box{{{0, 0, 0}}, {{1, 1, 1}}});
+  }
+  using memory_space = MemorySpace;
+};
+template <>
+struct Access<NearestToOrigin, PredicatesTag>
+{
+  KOKKOS_FUNCTION static std::size_t size(NearestToOrigin) { return 1; }
+  KOKKOS_FUNCTION static auto get(NearestToOrigin d, std::size_t)
+  {
+    return nearest(Point{0, 0, 0}, d.k);
+  }
+  using memory_space = MemorySpace;
+};
+} // namespace Traits
+} // namespace ArborX
+
+struct PairIndexDistance
+{
+  int index;
+  float distance;
+};
+
+struct PrintfCallback
+{
+  template <typename Predicate, typename OutputFunctor>
+  KOKKOS_FUNCTION void operator()(Predicate, int primitive,
+                                  OutputFunctor const &out) const
+  {
+    printf("Found %d from functor\n", primitive);
+    out(primitive);
+  }
+  template <typename Predicate, typename OutputFunctor>
+  KOKKOS_FUNCTION void operator()(Predicate, int primitive, float distance,
+                                  OutputFunctor const &out) const
+  {
+    printf("Found %d with distance %.3f from functor\n", primitive, distance);
+    out({primitive, distance});
+  }
+};
+
+int main(int argc, char *argv[])
+{
+  Kokkos::ScopeGuard guard(argc, argv);
+
+  int const n = 100;
+  std::vector<ArborX::Point> points;
+  // Fill vector with random points in [-1, 1]^3
+  std::uniform_real_distribution<float> dis{-1., 1.};
+  std::default_random_engine gen;
+  auto rd = [&]() { return dis(gen); };
+  std::generate_n(std::back_inserter(points), n, [&]() {
+    return ArborX::Point{rd(), rd(), rd()};
+  });
+
+  ArborX::BVH<MemorySpace> bvh{
+      ExecutionSpace{},
+      Kokkos::create_mirror_view_and_copy(
+          MemorySpace{},
+          Kokkos::View<ArborX::Point *, Kokkos::HostSpace,
+                       Kokkos::MemoryUnmanaged>(points.data(), points.size()))};
+
+  {
+    Kokkos::View<int *, MemorySpace> values("values", 0);
+    Kokkos::View<int *, MemorySpace> offsets("offsets", 0);
+    bvh.query(ExecutionSpace{}, FirstOctant{}, PrintfCallback{}, values,
+              offsets);
+#ifndef __NVCC__
```

Comment:
This seems to be unusual check. We usually protect with something like `__CUDA__`, but I guess you checked that it works with Clang+Cuda?

Reply:
Yes it is only a limitation with NVCC